algorithms/ml/nearest_neighbor.py

```diff
@@ -0,0 +1,41 @@
+import numpy
```

Comment:
We are using a 3rd party module here. `numpy` will become this repo's requirement/dependency, which is not desirable. I think we can implement these without using `numpy`, unless it is absolutely necessary (and I think that won't be the case), as these are meant to be _minimal_ algorithms in Python.  
By the way, great idea to implement ML algorithms here. I will also add some when I get time:)

Reply:
Here free online course on edx : https://www.edx.org/course/machine-learning-fundamentals-uc-san-diegox-dse220x